sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientAsyncCodeSnippets.java

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+
+
+public class TableClientAsyncCodeSnippets {
+
+
+    public void AsyncSnippets() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        // Add a table
+        tableServiceAsyncClient.createTable("OfficeSupplies").subscribe(Void -> {
+            System.out.println("Table creation successful.");
+        }, error -> {
+            System.out.println("There was an error creating the table. Error: " + error);
+        });
+
+
+        // Delete a table
+        tableServiceAsyncClient.deleteTable("OfficeSupplies").subscribe(Void -> {
+            System.out.println("Table deletion successful");
+        }, error -> {
+            System.out.println("There was an error deleting the table. Error: " + error);
+        });
+
+
+        // Query tables
+        String filterString = "$filter=TableName eq 'OfficeSupplies'";
+        tableServiceAsyncClient.queryTables(null, null, filterString).subscribe(azureTable -> {
```

Comment:
'null', 'null' is weird to see as the first parameters. I usually organize my parameters by how important or mandatory they are. In this case, `top` and `select` are options. How about an options class?
Is `filterString` mandatory?

```java
class TableQueryOptions {
    TableQueryOptions setTop(int top);
    TableQueryOptions setSelect(String select);
}

class TableServiceAsyncClient {
    PagedFlux<TableEntity> queryTables(String filter);
    PagedFlux<TableEntity> queryTables(String filter, TableQueryOptions queryOptions);
}
```

So your snippet would look like:
```java
tableServiceAsyncClient.queryTables(filterString);